### ISSUE TITLE
firmware: os-cpu-microcode plugins cross install will force the other one as missing

### ISSUE THREAD

**Issue by fichtner** [opnsense/plugins#4202]
**Important notices**

Our forum is located at https://forum.opnsense.org , please consider joining discussions there in stead of using GitHub for these matters.

Before you ask a new question, we ask you kindly to acknowledge the following:

- [x] I have read the contributing guide lines at https://github.com/opnsense/core/blob/master/CONTRIBUTING.md
- [x] I am convinced that my issue is new after having checked both open and closed issues at https://github.com/opnsense/core/issues?q=is%3Aissue

Minor annoyance is that installing Intel version first then AMD will mark the Intel one as missing instead of removing it from the config.xml. It doesn't know any better. Best to add more glue to register.php to handle such edge cases like we have with opnsense/core#7195 now  anyway.

**Comment by doktornotor** [comment]
Ok, vaguely recall AMD did not require the `cpu_microcode_name` but maybe it does [for the early update](https://reviews.freebsd.org/D44025) now...

I'll leave all those things there for now for testing some magic hackery. Cheers! 😉

**Comment by fichtner** [comment]
> Ok, vaguely recall AMD did not require the cpu_microcode_name but maybe it does [for the early update](https://reviews.freebsd.org/D44025) now...

Correct. We have a testing branch and kernel for that: https://github.com/opnsense/src/tree/amd_early

**Comment by fichtner** [comment]
First bit is 5d346589ed4431a048aeece4840c75d4bac9d753, now back to core lol
